pkg/controller/actions/status/releases/action_fetch_releases_status.go

```diff
@@ -0,0 +1,113 @@
+package releases
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/blang/semver/v4"
+	"github.com/opendatahub-io/opendatahub-operator/v2/apis/common"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/operator-framework/api/pkg/lib/version"
+	"gopkg.in/yaml.v3"
+	"path/filepath"
+)
+
+const (
+	ComponentMetadataFilename = "component_metadata.yaml"
+)
+
+// ComponentReleasesMeta represents the metadata for releases in the component_metadata.yaml file.
+// It contains a list of releases associated with a specific component.
+type ComponentReleasesMeta struct {
+	Releases []ComponentReleaseMeta `yaml:"releases,omitempty"`
+}
+
+// ComponentReleaseMeta represents the metadata of a single release within the component_metadata.yaml file.
+// It includes the release name, version, and repository URL.
+type ComponentReleaseMeta struct {
+	Name    string `yaml:"name,omitempty"`
+	Version string `yaml:"version,omitempty"`
+	RepoURL string `yaml:"repoUrl,omitempty"`
+}
+
+type Action struct {
+	labels map[string]string
+}
+
+type ActionOpts func(*Action)
+
+// run executes the reconciliation logic for fetching and processing component releases.
+//
+// This function:
+// 1. Reads the metadata file for the specified component.
+// 2. Parses the metadata file to extract release information.
+// 3. Updates the component's release status in the reconciliation request.
+//
+// Parameters:
+// - ctx: The context for managing deadlines and cancellations.
+// - rr: The reconciliation request containing the component instance.
+//
+// Returns:
+// - An error if the operation fails.
+func (a *Action) run(ctx context.Context, rr *types.ReconciliationRequest) error {
+	// Ensure the resource implements the WithReleases interface
+	obj, ok := rr.Instance.(common.WithReleases)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a WithReleases", rr.Instance)
+	}
+
+	// Build the path to the component metadata file
+	controllerName := strings.ToLower(rr.Instance.GetObjectKind().GroupVersionKind().Kind)
+	metadataPath := filepath.Join(odhdeploy.DefaultManifestPath, controllerName, ComponentMetadataFilename)
+
+	// Read the YAML file
+	yamlData, err := os.ReadFile(metadataPath)
+	if err != nil {
+		if os.IsNotExist(err) {
```

Comment:
is the absence of the metadata file an error ?

Reply:
I wasn’t sure what to return, but since the field isn’t mandatory, let's return `nil`.